src/WebJobs.Extensions.DurableTask/Listener/FunctionExecutionHelper.cs

```diff
@@ -0,0 +1,144 @@
+            IApplicationLifetimeWrapper hostServiceLifetime,
+            TriggeredFunctionData triggerInput,
+            CancellationToken cancellationToken)
+        {
+#pragma warning disable CS0618 // InvokeHandler approved for use by this extension
+            if (triggerInput.InvokeHandler == null)
+            {
+                throw new ArgumentException(
+                    $"{nameof(ExecuteFunctionInOrchestrationMiddleware)} should only be used when ${nameof(triggerInput)} has a value for ${nameof(TriggeredFunctionData.InvokeHandler)}");
+            }
+
+            try
+            {
+                bool executedUserCode = false;
+                var triggeredFunctionData = new TriggeredFunctionData()
+                {
+                    TriggerValue = triggerInput.TriggerValue,
+                    ParentId = triggerInput.ParentId,
+                    InvokeHandler = async userCodeHandler =>
+                    {
+                        executedUserCode = true;
+                        await triggerInput.InvokeHandler(userCodeHandler);
+                    },
+#if !FUNCTIONS_V1
+                    TriggerDetails = triggerInput.TriggerDetails,
+#endif
+                };
+#pragma warning restore CS0618
+
+                FunctionResult result = await executor.TryExecuteAsync(triggerInput, cancellationToken);
+                if (!result.Succeeded)
+                {
+                    if (!executedUserCode &&
+                         IsHostStopping(hostServiceLifetime))
+                    {
+                        return new WrappedFunctionResult(
+                            WrappedFunctionResult.FunctionResultStatus.FunctionsRuntimeError,
+                            result.Exception);
+                    }
+                    else
+                    {
+                        return new WrappedFunctionResult(
+                            WrappedFunctionResult.FunctionResultStatus.UserCodeError,
+                            result.Exception);
+                    }
+                }
+                else
+                {
+                    return new WrappedFunctionResult(
+                        WrappedFunctionResult.FunctionResultStatus.Success,
+                        null);
+                }
+            }
+            catch (Exception e)
+            {
+                return new WrappedFunctionResult(
+                     WrappedFunctionResult.FunctionResultStatus.FunctionsRuntimeError,
+                     e);
+            }
+        }
+
+        public static async Task<WrappedFunctionResult> ExecuteFunction(
+            ITriggeredFunctionExecutor executor,
+            IApplicationLifetimeWrapper hostServiceLifetime,
+            TriggeredFunctionData triggerInput,
+            CancellationToken cancellationToken)
+        {
+#pragma warning disable CS0618 // InvokeHandler approved for use by this extension
+            //if (triggerInput.InvokeHandler != null)
```

Comment:
This commented out code only needs to be in place if this method is only used by activities. In the current iteration, it is used for all function executions.

Reply:
I will uncomment it out considering we are going to keep the executedUserCode logic for orchestrations/entities, so this method should only be used for activities.